src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -164,7 +167,14 @@ MlirConvArgs MakeMlirConvArgs(const std::vector<size_t>& in_dims,
     StridedMemRef5D output{nullptr, nullptr, 0, {0, 0, 0, 0, 0}, {0, 0, 0, 0, 0}};
     initDimStrides(out_dims, out_strides, output);
 
-    return {filter, input, output};
+    if(populateWorkspaceArg)
+    {
+        StridedMemRef5D workspace{nullptr, nullptr, 0, {0, 0, 0, 0, 0}, {0, 0, 0, 0, 0}};
+        initDimStrides(weights_dims, weights_strides, workspace);
+        return {filter, input, output, workspace};
+    }
+
+    return {filter, input, output, {}};
```

Comment:
@atamazov It's used here.

Reply:
The optional `StridedMemRef5D` is only used by `MakeMlirWrWInvokerFactory` when invoking `MakeMlirConvArgs`. Other invokers would not use the field.

Reply:
@whchung But this simply calls a default constructor for the object. I do not see where we check if `workspace` exists using the `boost::optional` methods, like:
```C++
if(args.workspace)
{
    // Workspace exists; do something with *(args.workspace)
}
else
{
    // Optional args.workspace object does not exist and shouldn't be accessed
}
```
I recommend removing `boost::optional` if it is not used.

Reply:
@junliume FYI I have limited privileges and unable to "unresolve" comments.

Reply:
@atamazov I'm following the convention used in other parts of MIOpen when it comes to `boost::optional`. For example:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/target_properties.cpp#L82

I can remove the usage of `boost::optional` if you find it necessary. Please let me know if it's a recommendation or a requirement. Thanks.

Reply:
@atamazov Also, we don't need to check for `args.workspace` because of:
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1443/files#diff-6a213cde36650f2a3771d5e595c4f5754984da9a3ba95e32e5d2634846d6fadcR364-R375
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1443/files#diff-6a213cde36650f2a3771d5e595c4f5754984da9a3ba95e32e5d2634846d6fadcL344-R391

Reply:
I have just submitted a new commit to remove `boost::optional`.

Reply:
@whchung  
> @atamazov I'm following the convention used in other parts of MIOpen when it comes to `boost::optional`. For example: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/target_properties.cpp#L82

This is how we construct an optional object (and construction is done correctly in this PR). However if you construct an optional object, then let's use the "optional" semantics (otherwise we get overcomplicated code that looks suspicious).

> I can remove the usage of `boost::optional` if you find it necessary. Please let me know if it's a recommendation or a requirement. Thanks.

If optional semantics is not used, then it should be removed. In principle, this is a requirement, but not a hard one. For example, if it is difficult for you to rework the code, then I can do this for you, now or later.

@junliume can dismiss my blocking review, if necessary.